src/java/org/apache/cassandra/cql3/constraints/ScalarColumnConstraint.java

```diff
@@ -125,11 +127,18 @@ protected void internalEvaluate(AbstractType<?> valueType, ByteBuffer columnValu
     @Override
     public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
     {
+        returnType = columnMetadata.type;
+
+        if (!columnMetadata.name.equals(columnName))
+            throw new InvalidConstraintDefinitionException(String.format("Parameter of %s constraint should be the column name (%s)",
+                                                                         name(),
+                                                                         columnMetadata.name));
+
         validateTypes(columnMetadata);
 
         try
         {
-            value = columnMetadata.type.fromString(ParseUtils.unquote(term));
+            value = returnType.fromString(ParseUtils.unquote(term));
```

Comment:
not related to this PR directly but we might "cache" unquoted `term` so we don't do it all over again every time